client/my-sites/plans/ecommerce-trial/ecommerce-trial-expired/index.tsx

```diff
@@ -73,10 +76,15 @@ const ECommerceTrialExpired = (): JSX.Element => {
 		[ page, recordTracksEvent, settingsDeleteSiteUrl ]
 	);
 
+	const isWooExpressTrial = purchase?.isWooExpressTrial;
+	const isEntrepreneurTrial = purchase?.isWooExpressTrial === false;
```

Comment:
Here we want explicitly compare with false boolean, because to avoid the falsy comparison with undefined while purchases are not loaded yet.

Reply:
I see, not the ideal user experience but this should be required for the transition period only (2 weeks). After that we can clean this code up. Can we create a GH under the project so that we remember to clean this up. 🙂

Reply:
Yeah, maybe it is better to just add the spinner anyway. This experience will start when we merge it, a bit before the transition period. And yes we intend to clean up this and more on the plans page after the transition period.

![image](https://github.com/Automattic/wp-calypso/assets/33497086/07372d4f-b4bf-4479-bbeb-dff4313d8ae9)

Reply:
I created a task on the board to track all cleanups we need to do for the project.